### PR TITLE
Hook up the Settings UI JSON button

### PR DESCRIPTION
**Pull request by leonMSFT**
This PR will make the "Open JSON file" navigation view item in the Settings UI will open the user's settings file. Normally, `NavigationViewItems` raise their parent's `ItemInvoked` handler, but when a `NavViewItem` is in the `NavigationView.PaneFooter`, it doesn't do that. So, we have to hook up `OpenJson`'s `Tapped` and `KeyDown` handler to detect both a tap event and the enter keydown events.

References #1564

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -2571,7 +2571,15 @@ namespace winrt::TerminalApp::implementation
         // If we're holding the settings tab's switch command, don't create a new one, switch to the existing one.
         if (!_switchToSettingsCommand)
         {
-            auto newTabImpl = winrt::make_self<SettingsTab>(_settings);
+            winrt::Microsoft::Terminal::Settings::Editor::MainPage sui{ _settings };
+            sui.OpenJson([weakThis{ get_weak() }](auto&& /*s*/, auto&& /*e*/) {
+                if (auto page{ weakThis.get() })
+                {
+                    page->_LaunchSettings(SettingsTarget::SettingsFile);
```
Comment (leonMSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp:2578]: How should we let users open defaults.json?

Comment (DHowett) [reply src/cascadia/TerminalApp/TerminalPage.cpp:2578]: We could do the same Alt trick

**Review by miniksa — APPROVED** [review]

**Comment by miniksa** [comment]
Approved presuming you're gonna rev it once more with the Alt trick per Dustin's comment and be done there.

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 2]
```diff
@@ -2571,7 +2571,15 @@ namespace winrt::TerminalApp::implementation
         // If we're holding the settings tab's switch command, don't create a new one, switch to the existing one.
         if (!_switchToSettingsCommand)
         {
-            auto newTabImpl = winrt::make_self<SettingsTab>(_settings);
+            winrt::Microsoft::Terminal::Settings::Editor::MainPage sui{ _settings };
+            sui.OpenJson([weakThis{ get_weak() }](auto&& /*s*/, winrt::Microsoft::Terminal::Settings::Model::SettingsTarget e) {
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:2575]: clever

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 3]
```diff
@@ -236,4 +238,25 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+        const bool altPressed = WI_IsFlagSet(lAltState, CoreVirtualKeyStates::Down) ||
+                                WI_IsFlagSet(rAltState, CoreVirtualKeyStates::Down);
+
+        const auto target = altPressed ? SettingsTarget::DefaultsFile : SettingsTarget::SettingsFile;
+        _OpenJsonHandlers(nullptr, target);
+    }
+
+    void MainPage::OpenJsonKeyDown(IInspectable const& /*sender*/, Windows::UI::Xaml::Input::KeyRoutedEventArgs const& args)
+    {
+        if (args.Key() == VirtualKey::Enter)
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:256]: nit: do we need space here too?

Comment (leonMSFT) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:256]: hmm i don't see a space here? also runformat didn't seem to change anything 🤷

Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:256]: oh no, I mean `Spacebar` hahaha

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:256]: i think he means, "should Space activate the Open Json button"

Comment (leonMSFT) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:256]: LOL yes, yes it should

**Review by carlos-zamora — APPROVED** [review]
@msftbot merge this in 5 minutes

**Comment by carlos-zamora** [comment]
@msftbot merge this in 5 minutes

**Pull request merged**